afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/websocket/AutoReconnectionConnectionManager.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.ws.storage.websocket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.websocket.Session;
+import java.net.URI;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A connection manager which will attempt to reconnect at fixed time intervals.
+ *
+ * @author Sylvain Leclerc <sylvain.leclerc@rte-france.com>
+ */
+public class AutoReconnectionConnectionManager extends StandardConnectionManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AutoReconnectionConnectionManager.class);
+
+    private final ScheduledExecutorService scheduler;
+    private final long delay;
+    private final TimeUnit timeUnit;
+
+    private boolean closed = false;
+
+    public AutoReconnectionConnectionManager(URI uri, long delay, TimeUnit timeUnit) {
+        super(uri);
+        this.scheduler = Executors.newSingleThreadScheduledExecutor();
+        this.delay = delay;
+        this.timeUnit = timeUnit;
+    }
+
+    public AutoReconnectionConnectionManager(URI uri, long delayInSeconds) {
+        this(uri, delayInSeconds, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void onClose(Session session, Object clientEndoint) {
+        if (!closed) {
+            LOGGER.warn("Connection to {} has been closed, will try to reconnect in {}", getUri(), getDelayAsString());
+            scheduleReconnection(clientEndoint);
+        }
+    }
+
+    private void scheduleReconnection(Object clientEndoint) {
+        scheduler.schedule(() -> attemptReconnection(clientEndoint), delay, timeUnit);
```

Comment:
Sometimes, the reconnection delay is longer after several retries. In your proposal, we could consider as a DoS attack?

Reply:
You mean, you would prefer a more complex logic with an increasing interval ?

Maybe it's better, I just wanted to make things simple at first, since for my use case, it seems OK to have fixed intervals:
 * we would configure a large enough interval (1minute maybe) so that it does not "spam" too much.
Basically the failed attempts will be when the server is off, so from its point of view it will not be an issue anyway :) 
 * if the interval gets too large and we restart the server only after a long time, it can be bothering to need to wait for example 1 hour before the reconnection

Reply:
I agree to keep it like this as a first approach and let the responsibility to the user to set a value big enough. Maybe we have to well explain how this will spam the server, and suggest a "good" value for this parameter.

Do you know the value of the connection timeout you use? The default TCP value is usually 5 or 30 seconds.

Reply:
I must say I don't know how the websocket connection is kept alive, in my case. It does not seem to be broken by any TCP timeout, while I am pretty sure there are periods of inactivity longer than 30s. Might be part of the websocket protocol to have some mechanism to keep it alive ? Will investigate, this, some day :)